### DISCUSSION TITLE
Monthly Future Regressors

### DISCUSSION THREAD

**Discussion by winnieshen96** [ourownstory/neural_prophet#607] [Q&A - get help using NeuralProphet] [answered]
In the original Prophet documentation, when modeling monthly data forecasts, according to [link here](https://facebook.github.io/prophet/docs/non-daily_data.html), I need to add 12 binary variables to the model and not use yearly seasonality to avoid overfitting. Is neuralprophet recommending the same approach to model monthly data? However as I add the 12 binary variables as future regressors, I encounter this bug 
`ERROR - (NP.df_utils.auto_normalization_setting) - encountered variable with one unique value`
And also when I am plotting the components of the model the output is strange too.

**Describe the solution you'd like**

A documentation of tutorial in dealing with sub-daily data

**Describe alternatives you've considered**

Use yearly seasonality

**Additional context**

No

**Comment by ourownstory** [comment, marked as answer]
Hi @winnieshen96 
Thank you for raising this Question.

The simplest approach would be to set `yearly_seasonality` to a lower value.
The model will have `2*yearly_seasonality` additional parameters (compared to no seasonality).
Thus, setting it to a low value such as 2-6 should not lead to overfitting, even if you only have a few years of data.

The second approach of creating a regressor per month is potentially even better, but requires some extra work.
The Error message you see is expected in certain cases, but should not appear here.
Do you mind sharing a minimal code example to help us reproduce and address the bug?
Thank you!

**Reply by ourownstory** [reply]
@winnieshen96 
I added two issues #608 and #608 to support your suggested feature.
